library/forms/lib/components/FormManager/index.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import type {ComponentProps} from 'react';
+import type {EncodedNotebook} from '@faims3/data-model';
+import {FormSection} from '../FormSection';
+import {FormContext, FormContextType} from './FormContext';
+
+const formValues = {
+  'Full-Name': 'Steve',
+  Occupation: 'Developer',
+};
+
+export interface FormManagerProps extends ComponentProps<any> {
+  project: EncodedNotebook;
+  formName: string;
+}
+
+export const FormManager = (props: FormManagerProps) => {
+  console.log('FormManager:', props.formName);
+  const [currentValues, setCurrentValues] =
+    React.useState<Record<string, any>>(formValues);
+
+  // stable reference to setValue function used by fields to update
+  // the value of a field in the form state
+  const setValue = React.useCallback((fieldName: string, value: any) => {
+    console.log(`Set field ${fieldName} to value ${value}`);
+    setCurrentValues(prev => ({
+      ...prev,
+      [fieldName]: value,
+    }));
+  }, []);
+
+  // Stable reference to current values
+  const valuesRef = React.useRef(currentValues);
+  valuesRef.current = currentValues;
```

Comment:
this will be run on every re-render, so yeah I think there's no point to this layer